### PR TITLE
virtctl: init at 0.52.0

### PR DESCRIPTION
**Pull request by haslersn**
###### Description of changes

KubeVirt's client tool: https://kubevirt.io/user-guide/operations/virtctl_client_tool/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/virtualization/kubevirt/virtctl.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  tags = [ "selinux" ];
+
+  ldflags = let
+    gitCommit = "836afc8c556206fc7715615ddb6516d76f4240db";
+  in [
+    "-X kubevirt.io/client-go/version.gitCommit=${gitCommit}"
+    "-X kubevirt.io/client-go/version.gitTreeState=clean"
+    "-X kubevirt.io/client-go/version.gitVersion=v${version}"
+  ];
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/kubevirt/virtctl.nix:32]:
```suggestion
  ldflags = [
    "-X kubevirt.io/client-go/version.gitCommit=v${version}"
    "-X kubevirt.io/client-go/version.gitTreeState=clean"
    "-X kubevirt.io/client-go/version.gitVersion=v${version}"
  ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/virtualization/kubevirt/virtctl.nix (hunk 1 above)
Comment (haslersn) [reply pkgs/tools/virtualization/kubevirt/virtctl.nix:32]: I like this, cause it's less maintenance.

It's not consistent with the officially released precompiled binary by KubeVirt. Should I apply this change anyway?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -1203,6 +1203,8 @@ with pkgs;
 
   ventoy-bin = callPackage ../tools/cd-dvd/ventoy-bin { };
 
+  virtctl = callPackage ../tools/virtualization/kubevirt/virtctl.nix { };
```
[the remark this answers is not in the document]
Comment (haslersn) [reply pkgs/top-level/all-packages.nix:1206]: Should I change the package name to `kubevirt` then? This is only virtctl and there are other KubeVirt components, too. The other KubeVirt components never need to be installed on NixOS, as they only run in containers. But in theory these containers could one day be built with Nix (the package manager).

File: pkgs/tools/virtualization/kubevirt/virtctl.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{ buildGo117Module
+, fetchFromGitHub
+, lib
+, testers
+, virtctl
+}:
+
+buildGo117Module rec {
```
[the remark this answers is not in the document]
Comment (haslersn) [reply pkgs/tools/virtualization/kubevirt/virtctl.nix]: I thought `buildGoModule` defaults to Go 1.16. But apparently now it's 1.17. Was this changed recently?

I applied your suggestion.

File: pkgs/tools/virtualization/kubevirt/virtctl.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/virtualization/kubevirt/virtctl.nix:32]: > It's not consistent with the officially released precompiled binary by KubeVirt. Should I apply this change anyway?

Fine for me, I don't really care about the commit personally.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix]: Oh, right. Makes sense.

File: pkgs/tools/virtualization/kubevirt/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  passthru.tests.version = testers.testVersion {
+    package = kubevirt;
+    command = "virtctl version --client";
+    version = "v${version}";
+  };
+
+  meta = with lib; {
+    description = "Client tool to use advanced features such as console access";
+    homepage = "https://kubevirt.io/";
+    license = licenses.apsl20;
```
Comment (fabaff) [inline pkgs/tools/virtualization/kubevirt/default.nix:41]:
```suggestion
    license = licenses.asl20;
```

The license is [Apache License 2.0](https://github.com/kubevirt/kubevirt/blob/main/LICENSE) not "Apple Public Source License 2.0".

Comment (haslersn) [reply pkgs/tools/virtualization/kubevirt/default.nix:41]: Oops.

Comment (haslersn) [reply pkgs/tools/virtualization/kubevirt/default.nix]: Fixed

File: pkgs/tools/virtualization/kubevirt/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+, fetchFromGitHub
+, lib
+, testers
+, kubevirt
+}:
+
+buildGoModule rec {
+  pname = "kubevirt";
+  version = "0.52.0";
+  strictDeps = true;
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/kubevirt/default.nix:11]:
```suggestion
```
Thats the default already https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/go-modules/generic/default.nix#L263

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**